pkg/lvm/lvm_util.go

```diff
@@ -507,6 +543,132 @@ func ListLVMVolumeGroup() ([]apis.VolumeGroup, error) {
 	return decodeVgsJSON(output)
 }
 
+//Function to get LVM Logical volume device
+//It returns LVM logical volume device(dm-*).
+//This is used as a label in metrics(lvm_lv_total_size) which helps us to map lv_name to device.
+//
+//Example: pvc-f147582c-adbd-4015-8ca9-fe3e0a4c2452(lv_name) -> dm-0(device)
+func getLvDeviceName(path string) (string, error) {
+	dmPath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		klog.Errorf("failed to resolve device mapper from lv path %v: %v", path, err)
+		return "", err
+	}
+	deviceName := strings.Split(dmPath, "/")
+	return deviceName[len(deviceName)-1], nil
+}
+
+//To parse the output of lvs command and store it in LogicalVolume
+//It returns LogicalVolume.
+//
+//Example: LogicalVolume{
+//		Name:     "pvc-213ca1e6-e271-4ec8-875c-c7def3a4908d",
+//		FullName: "linuxlvmvg/pvc-213ca1e6-e271-4ec8-875c-c7def3a4908d",
+//		UUID:     "UJp2Dh-Knfo-E0fO-KjPB-RSHO-X7JO-AI2FZW",
+//		Size:     3221225472,
+//		Path:     "/dev/linuxlvmvg/pvc-213ca1e6-e271-4ec8-875c-c7def3a4908d",
+//		DMPath:   "/dev/mapper/linuxlvmvg-pvc--213ca1e6--e271--4ec8--875c--c7def3a4908d",
```

Comment:
can we add Device also in the comment?

Reply:
Device is not getting populated in this function. I have mentioned it over here [comment](https://github.com/openebs/lvm-localpv/pull/94/files/63094c17e05b44ab97dc5fc94be72042ff7407fe#diff-1d348d11d677906b0fb0a0506d50f2bfed43754cc0f16b6509a3bef047095722R621)